src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -264,20 +264,64 @@
                :message (str "Invalid service_status_version. Should be an "
                           "integer but was " level)}))))
 
+(schema/defn ^:always-validate summarize-states
+  "Given a map of service statuses:
+   * if all of the statuses have the same :state, returns that :state
+   * if not all of the statuses are the same, returns :unknown"
+  [statuses :- ServicesStatus] :- State
+  (let [state-set (->> statuses
+                       vals
+                       (map :state)
+                       set)]
+    (if (= 1 (count state-set))
+      (first state-set)
+      :unknown)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Compojure App
 
-(defn build-routes
+(schema/defn ^:always-validate status->code
+  "Given a service status, returns an appropriate HTTP status code"
+  [status :- ServiceStatus] :- schema/Int
+  (if (nominal? status) 200 503))
+
+(schema/defn ^:always-validate statuses->code
+  "Given a map of service statuses, returns an appropriate HTTP status code."
+  [statuses :- ServicesStatus] :- schema/Int
+  (if (all-nominal? statuses)
+    200
+    503))
```

Comment:
Why three lines for the if here but all on one line above? The extra `all-` in the second example doesn't put it anywhere close to the line limit.